### ISSUE TITLE
Angular does not build

### ISSUE THREAD

**Issue by HuiiBuh** [georgebyte/rxjs-observable-store#8]
I installed your library and after the installation angular does not build any more (stops and does not continue without error). 

The problem seems to be *ts-toolbelt* and everything which looks like this. 
 
```typescript
 // prettier-ignore
    patchState<
        P1 extends keyof Object.Path<S, []>
    >(value: Object.Path<S, [P1]>, part1: P1): void;
```
Once I removed all this the library worked again. 

I am using angular  10.0.5 with  node 12.16.1

**Comment by georgebyte** [comment]
Hi, thanks for reporting the issue. I had a look at it and the problem seems to be with Angular@10.x using TypeScript version `~3.9.5`. Angular@9.x uses TS version `~3.8.3` which works fine. I haven't been able to find a solution yet, at least not one which would provide the same level of type safety and autocompletion of path params as the current implementation does.
I'll update this issue with my progress.

**Comment by ealib** [comment]
With TS version 3.9.7 you get

```bash
ERROR in node_modules/rxjs-observable-store/lib/esm/store.d.ts:19:556 - error TS2589: Type instantiation is excessively deep and possibly infinite.
    
    19     patchState<P1 extends keyof Object.Path<S, []>, P2 extends keyof Object.Path<S, [P1]>, P3 extends keyof Object.Path<S, [P1, P2]>, P4 extends keyof Object.Path<S, [P1, P2, P3]>, P5 extends keyof Object.Path<S, [P1, P2, P3, P4]>, P6 extends keyof Object.Path<S, [P1, P2, P3, P4, P5]>, P7 extends keyof Object.Path<S, [P1, P2, P3, P4, P5, P6]>, P8 extends keyof Object.Path<S, [P1, P2, P3, P4, P5, P6, P7]>, P9 extends keyof Object.Path<S, [P1, P2, P3, P4, P5, P6, P7, P8]>, P10 extends keyof Object.Path<S, [P1, P2, P3, P4, P5, P6, P7, P8, P9]>>(value: Object.Path<S, [P1, P2, P3, P4, P5, P6, P7, P8, P9, P10]>, part1: P1, part2: P2, part3: P3, part4: P4, part5: P5, part6: P6, part7: P7, part8: P8, part9: P9, part10: P10): void;
       


                                                                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    node_modules/rxjs-observable-store/lib/esm/store.d.ts:30:672 - error TS2589: Type instantiation is excessively deep and possibly infinite.

    30     onChanges<P1 extends keyof Object.Path<S, []>, P2 extends keyof Object.Path<S, [P1]>, P3 extends keyof Object.Path<S, [P1, P2]>, P4 extends keyof Object.Path<S, [P1, P2, P3]>, P5 extends keyof Object.Path<S, [P1, P2, P3, P4]>, P6 extends keyof Object.Path<S, [P1, P2, P3, P4, P5]>, P7 extends keyof Object.Path<S, [P1, P2, P3, P4, P5, P6]>, P8 extends keyof Object.Path<S, [P1, P2, P3, P4, P5, P6, P7]>, P9 extends keyof Object.Path<S, [P1, P2, P3, P4, P5, P6, P7, P8]>, P10 extends keyof Object.Path<S, [P1, P2, P3, P4, P5, P6, P7, P8, P9]>>(part1: P1, part2: P2, part3: P3, part4: P4, part5: P5, part6: P6, part7: P7, part8: P8, part9: P9, part10: P10): Observable<Object.Path<S, [P1, P2, P3, P4, P5, P6, P7, P8, P9, P10]>>;
       
```

**Comment by georgebyte** [comment]
The issue is fixed in `v3.0.4`.

**Comment by HuiiBuh** [comment]
Nice thanks a lot
